### PR TITLE
Log the missing static training memory estimate

### PR DESCRIPTION
**Pull request by infinuts**
Adds one static training memory estimate to the existing `DebugCallback` at training start.

The calculation sums parameter storage and assumes FP32 gradients plus two FP32 Adam moments for trainable parameters. It uses `ds_numel` for ZeRO-3 parameters so empty local partitions do not produce a zero estimate.

The log labels this as an unsharded estimate and lists what it excludes, including activations, master weights, quantization metadata, reference models, and rollout caches. Sharding and offload affect actual per-GPU usage.

The diff is limited to `utils/debug_callback.py` and CPU tests for the calculation, startup call, and failure handling.

Validation: `python -m pytest -q -rs` passed with 135 tests passing and one skipped because DeepSpeed is not installed. The ZeRO-3 calculation is tested with fake partitioned parameters.

### CODE REVIEW

**Comment by youngwan-roblox** [comment]
We are already printing the training hyperparameters. Would you add only the missing parameter? Most of the changes in your PR are duplicates.

I’ve attached an example log.
```
FAI-RL version: 0.2.18
Algorithm: SFT
Base model: Qwen/Qwen3.8-27B
Dataset: 
Region: us-east-1

Distributed training:
  world_size: 2
  DeepSpeed: ZeRO-3
  Device: CUDA
  dtype: bfloat16

Training:
  epochs: 1
  max_length: 4096
  train_batch_size: 4/device
  eval_batch_size: 1/device
  gradient_accumulation_steps: 2
  learning_rate: 1e-4
  warmup_steps: 2
  save_steps: 7
  eval_steps: 7
  eval_split_ratio: 0.1

LoRA:
  enabled: true
  r: 16
  alpha: 32
  dropout: 0.05
  target_modules: q_proj, k_proj, v_proj, o_proj, gate_proj, up_proj, down_proj

Optimization:
  gradient_checkpointing: true
  Flash Attention: true
  bf16: true

Evaluation / Early Stopping:
  eval_enabled: true
  metric_for_best_model: eval_loss
  early_stopping: true
  early_stopping_patience: 3
  early_stopping_threshold: 0.0

Checkpoint:
  upload_checkpoints: true
  upload_final_model: false
  save_only_model: true

**Comment by infinuts** [comment]
Thanks for pointing that out. I checked the existing callback and found it already logs effective batch size and trainable parameter counts too.

I've removed the separate summary and all six trainer hooks. The PR now adds only the missing static memory estimate to `DebugCallback`, with its assumptions in the log message. The diff is two files, including tests.

The full CPU suite passes: 135 passed, 1 skipped because DeepSpeed is not installed.

**Comment by youngwan-roblox** [comment]
thanks for the changes.  approved and merged.

**Review by youngwan-roblox — APPROVED** [review]
